### PR TITLE
[master] Fix image defaults/config

### PR DESCRIPTION
**Pull request by SahAssar**
Forward-port of #4973, I've tested that it produces the same results as 2.2.X does now. Ping @CarsonF and @GawainLynch for checking :smile:

### CODE REVIEW

**Comment by SahAssar** [comment]
Alright, this is going to need some tests updating that I didn't know about when I opened it...

**Comment by CarsonF** [comment]
Isn't this less intuitive?

**Comment by SahAssar** [comment]
CarsonF: It makes the defaults actually be defaults, instead of them being hardcoded. I don't think that will be less intuitive. For example, the defaults for the a thumbnail are the defaults that have been set, unless you use a popup in which case the defaults are 100x100, which can't be changed. Same goes for cropping on thumbnails. We have a default config so we should actually use it, right?

File: src/Twig/Handler/ImageHandler.php [hunk 1]
```diff
@@ -155,12 +155,20 @@ public function showImage($fileName = null, $width = null, $height = null, $crop
         }
         $thumb = $this->getThumbnail($fileName, $width, $height, $crop);
 
-        if ($width === null || $height === null) {
+        if ($width === null && $height === null) {
+            $thumbconf = $this->app['config']->get('general/thumbnails');
+            $width = !empty($thumbconf['default_image'][0]) ? $thumbconf['default_image'][0] : 1000;
+            $height = !empty($thumbconf['default_image'][1]) ? $thumbconf['default_image'][1] : 750;
+            $thumb->setWidth($width);
+            $thumb->setHeight($height);
```
Comment (GawainLynch) [inline src/Twig/Handler/ImageHandler.php]: I might be missing something here, but isn't this already done in `\Bolt\Helpers\Image\Thumbnail`? 

The point of that class was to abstract, and centralise all of this :wink:

Comment (SahAssar) [inline src/Twig/Handler/ImageHandler.php]: Yeah, but there it assumes that you want the default_thumbnail defaults, here we want the default_image defaults. This is basically the same thing that we already do in the `popup` function.

Comment (GawainLynch) [inline src/Twig/Handler/ImageHandler.php]: Well, yes. That would be why the function is called `showImage()`… :koala:

**Comment by CarsonF** [comment]
@SahAssar I meant having html in the yaml config :/

**Comment by SahAssar** [comment]
@CarsonF Not sure what you mean? There is no html in the yaml AFAIK... Also, this didn't add anything to the config, it only makes the image functions use the defaults set if one uses showimage etc. with no values set.

**Comment by CarsonF** [comment]
@SahAssar Nevermind I'm a derp. I misread your code block on the other issue as yaml instead of results. :+1: :koala:

**Comment by GawainLynch** [comment]
Always the damn :koala: :wink:

**Comment by SahAssar** [comment]
@CarsonF Ah, I probably shouldn't have had the config and the results in the same block :laughing:

**Pull request merged**